### DISCUSSION TITLE
nrspre hanging on an A100 system.

### DISCUSSION THREAD

**Discussion by jigo3635** [Nek5000/nekRS#311] [Compiling]
Hi,

By following the instruction in README.md,  the code can be compiled and binaries are built in the default folder

```
$ module add GCC CUDA OpenMPI
$ module add CMake
$ ./makers
$ ls ~/.local/nekrs/bine
nekrs               nrsbmpi  nrspre          nrsqsub_slurm   nrsvis  uncrustify.cfg  update3rdParty
nrsLaunchHelper.sh  nrsmpi   nrsqsub_lassen  nrsqsub_summit  occa    uncrustify.sh

```
But when I run the example, It is hanging forever until killing it.
```
$ export NEKRS_HOME=$HOME/.local/nekrs
$ export PATH=$NEKRS_HOME/bin:$PATH
$ cd $NEKRS_HOME/examples/turbPipePeriodic
(changed "mpirun -np 1" to  "srun -n 1" in nrspre)
$ salloc -N 1  --gres=gpu:4 ...
$ nrspre turbPipe 1
(hanging)
```

The system support said that probably the OCCA library is somewhat contradict against the `mpi-setting`  in the system. Do you have any experiences with the issue?

Thanks. /Jing

**Comment by stgeke** [comment]
What system is this? Can you please try to use strace to see where it hangs.
btw: occa doesn't interact with MPI

**Comment by jigo3635** [comment]
It is Juwels booster system at Julich. Command strace is not available on the compute nodes. I will try other tools. 

Thanks. /Jing
